promdump/promdump.go

```diff
@@ -368,7 +381,113 @@ func exportMetric(ctx context.Context, promApi v1.API, metric string, beginTS ti
 			}
 		}
 	}
-	return writeFile(&values, filePrefix, fileNum)
+
+	return fileNum, writeFile(&values, filePrefix, fileNum)
+
+}
+
+func createArchive(buf io.Writer) error {
+	// Create new Writers for gzip , bzip2 and tar
+	// These writers are chained. Writing to the tar writer will
+
+	var tw *tar.Writer
+
+	switch *compression {
+	case "gzip":
+		gw := gzip.NewWriter(buf)
+		defer gw.Close()
+		tw = tar.NewWriter(gw)
+
+	case "bzip2":
+		// Add support for bzip2 compression
+		bw, err := bzip2.NewWriter(buf, nil) // Pass nil as the second argument, since bzip2 required to two argument so
+		if err != nil {
+			return err
+		}
+		defer bw.Close()
+		tw = tar.NewWriter(bw)
+	default:
+		return fmt.Errorf("unsupported compression type: %s", *compression)
+
+	}
+	defer tw.Close()
+
+	var filesToArchive []string
+
+	// Iterate over collectMetrics and add them to the tar archive
+	for _, v := range collectMetrics {
+		for i := 0; i < int(v.fileCount); i++ {
+			metricName, err := getMetricName(v)
+			if err != nil {
+				log.Fatalf("createArchive: %v", err)
+			}
+			filename := fmt.Sprintf("%s.%05d", metricName, i)
+			filesToArchive = append(filesToArchive, filename)
+		}
+	}
+	//custom filename collect nd add them to the tar archiv
+	if *metric != "" {
+		for i := 0; i < int(customMatricCount); i++ {
+			filename := fmt.Sprintf("%s.%05d", *metric, i)
```

Comment:
The `*metric` value is the metric string we're passing to Prometheus to retrieve the custom metric. The output filename for custom metrics is stored in `*out`.